File: MessageBuilder.java
package net.glitchtechs.starapi;

import net.md_5.bungee.api.chat.BaseComponent;
import net.md_5.bungee.api.chat.ClickEvent;
import net.md_5.bungee.api.chat.HoverEvent;
import net.md_5.bungee.api.chat.TextComponent;
import net.md_5.bungee.api.chat.HoverEvent.Action;
import org.bukkit.entity.EntityType;

public class MessageBuilder {
    public static TextComponent buildHoverText(TextComponent baseComponent, String hoverText) {
        baseComponent.setHoverEvent(new HoverEvent(Action.SHOW_TEXT, new TextComponent[]{new TextComponent(hoverText)}));
        return baseComponent;
    }

    public static TextComponent buildCommandText(String text, String clickCommand) {
        TextComponent component = new TextComponent(text);
        component.setClickEvent(new ClickEvent(net.md_5.bungee.api.chat.ClickEvent.Action.RUN_COMMAND, clickCommand));
        return component;
    }

    public static TextComponent buildEntityText(String text, EntityType entityType) {
        TextComponent component = new TextComponent(text);
        component.setHoverEvent(new HoverEvent(Action.SHOW_ENTITY, new BaseComponent[]{new TextComponent("{id:\"minecraft:" + entityType.getKey().getKey() + "\"}")}));
        return component;
    }

    public static TextComponent buildOpenUrlText(String text, String url) {
        TextComponent component = new TextComponent(text);
        component.setClickEvent(new ClickEvent(net.md_5.bungee.api.chat.ClickEvent.Action.OPEN_URL, url));
        return component;
    }
}

File: Kotlin/src/MessageBuilder.kt
package net.glitchtechs.starapi

import net.md_5.bungee.api.chat.BaseComponent
import net.md_5.bungee.api.chat.ClickEvent
import net.md_5.bungee.api.chat.HoverEvent
import net.md_5.bungee.api.chat.TextComponent
import org.bukkit.entity.EntityType


object MessageBuilder {
    fun buildHoverText(baseComponent: TextComponent, hoverText: String?): TextComponent {
        baseComponent.hoverEvent = HoverEvent(HoverEvent.Action.SHOW_TEXT, arrayOf(TextComponent(hoverText)))
        return baseComponent
    }

    fun buildCommandText(text: String?, clickCommand: String?): TextComponent {
        val component = TextComponent(text)
        component.clickEvent = ClickEvent(ClickEvent.Action.RUN_COMMAND, clickCommand)
        return component
    }

    fun buildEntityText(text: String?, entityType: EntityType): TextComponent {
        val component = TextComponent(text)
        component.hoverEvent = HoverEvent(HoverEvent.Action.SHOW_ENTITY, arrayOf<BaseComponent>(TextComponent("{id:\"minecraft:" + entityType.key.key + "\"}")))
        return component
    }

    fun buildOpenUrlText(text: String?, url: String?): TextComponent {
        val component = TextComponent(text)
        component.clickEvent = ClickEvent(ClickEvent.Action.OPEN_URL, url)
        return component
    }
}